Documentation/ceph-osd-mgmt.md

```diff
@@ -0,0 +1,118 @@
+
+To add more OSDs, Rook will automatically watch for new nodes and devices being added to your cluster.
+If they match the filters or other settings in the `storage` section of the cluster CR, the operator
+will create new OSDs.
+
+## Add an OSD on a PVC
+
+In more dynamic environments where storage can be dynamically provisioned with a raw block storage provider, the OSDs can be backed
+by PVCs. See the `storageClassDeviceSets` documentation in the [Cluster CRD](ceph-cluster-crd.md#storage-class-device-sets) topic.
+
+To add more OSDs, you can either increase the `count` of the OSDs in an existing device set or you can
+add more device sets to the cluster CR. The operator will then automatically create new OSDs according
+to the updated cluster CR.
+
+## Remove an OSD
+
+Removal of OSDs is intentionally not automated. Rook's charter is to keep your data safe, not to delete it. If you are
+really sure you want or need to remove OSDs, it can be done. We just want you to be in control of this action.
+
+To remove an OSD due to a failed disk or other re-configuration, first you should verify that your cluster is not full and that
+the remaining OSDs and their placement groups (PGs) are healthy in order to handle the rebalancing of the data.
+If all the PGs are `active+clean` and there are no warnings about being low on space, this means the data is fully replicated
+and it is safe to proceed. If an OSD is failing, the PGs will not be perfectly clean and you will need to proceed anyway.
+
+### From the Toolbox
+1. Determine the OSD ID for the OSD to be removed. The osd pod may be in an error state such as `CrashLoopBackoff` or the `ceph` commands
+in the toolbox may show which OSD is `down`.
+2. Mark the OSD as `out` if not already marked as such by Ceph. This signals Ceph to start moving (backfilling) the data that was on that OSD to another OSD.
+   - `ceph osd out osd.<ID>` (for example if the OSD ID is 23 this would be `ceph osd out osd.23`)
+3. If desired, wait for the data to finish backfilling to other OSDs.
+   - `ceph status` will indicate the backfilling is done when all of the PGs are `active+clean`.
+4. Remove the disk from the node.
+5. Update your CephCluster CR such that the operator won't create an OSD on the device anymore.
+Depending on your CR settings, you may need to remove the device from the list or update the device filter.
+If you are using `useAllDevices: true`, no change to the CR is necessary.
+6. Remove the OSD from the Ceph CRUSH map
+   - `ceph osd crush remove osd.<ID>`
+7. Verify the OSD is removed from the node in the CRUSH map
+   - `ceph osd tree`
+8. Delete the Ceph authorization
+   - `ceph auth del osd.<ID>`
+9. Delete the OSD
+   - `ceph osd rm osd.<ID>`
+
+### Outside Ceph
+10. Delete the deployment:
+   - `kubectl delete deployment -n rook-ceph rook-ceph-osd-<ID>`
```

Comment:
Wouldn't the operator remove that deployment once the CR was edited? Or not anymore? (purpose of the PR?)

Reply:
I should mention the operator will do it automatically if the cluster CR has this set: `removeOSDsIfOutAndSafeToRemove: true`